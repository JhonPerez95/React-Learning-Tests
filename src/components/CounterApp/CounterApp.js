import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import useStyles from './style';

const CounterApp = ({ value }) => {
  // MATERIAL-UI
  const classes = useStyles();

  //STATE
  const [count, setCount] = useState(value);

  //ONCLICK
  const handlerCount = () => {
    setCount(count + value);
  };

  const handlerReset = () => {
    setCount(0);
  };

  const handlerCountRest = () => {
    setCount(count - value);
  };

  return (
    <>
      <h1>Counter App</h1>
      <h2>{count}</h2>
      <div className={classes.root}>
        <Button
          onClick={handlerCount}
          variant="contained"
          color="primary"
          id="btnIncrement"
        >
          Count + {value}
        </Button>
        <Button onClick={handlerReset} variant="contained" color="secondary">
          Reset
        </Button>
        <Button onClick={handlerCountRest} variant="contained">
          Count - {value}
        </Button>
      </div>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
