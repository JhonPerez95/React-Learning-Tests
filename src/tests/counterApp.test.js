import React from 'react';

import '@testing-library/jest-dom';
// import { render } from '@testing-library/react';
import CounterApp from '../components/CounterApp/CounterApp';
import { shallow } from 'enzyme';

describe('Pruebas al componente <CounterApp>', () => {
  // test('Debe enviar un numero por parametro Props', () => {
  //   const num = 4;
  //   const { getByText } = render(<CounterApp value={num} />);
  //   // expect(getByText(num)).toBeInTheDocument();
  // });

  let wrapper = shallow(<CounterApp value={5} />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp value={5} />);
  });

  test('Renderizado correctamente de <CounterApp/>', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Mostrar por defecto en valor 100 ', () => {
    const wrapper = shallow(<CounterApp value={100} />);
    const counterText = wrapper.find('h2').text().trim();
    expect(counterText).toBe('100');
  });

  test('debe de incrementar con el botÃ³n +1', () => {
    const btn = wrapper.find('div.button');
    console.log(btn.html());
    // const counterText = wrapper.find('h2').text().trim();
    // expect(counterText).toBe('6');
  });
});
