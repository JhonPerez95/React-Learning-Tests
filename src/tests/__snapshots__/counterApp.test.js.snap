// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <CounterApp> Renderizado correctamente de <CounterApp/> 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    5
  </h2>
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(Button))
      color="primary"
      id="btnIncrement"
      onClick={[Function]}
      variant="contained"
    >
      Count + 
      5
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="secondary"
      onClick={[Function]}
      variant="contained"
    >
      Reset
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      variant="contained"
    >
      Count - 
      5
    </WithStyles(ForwardRef(Button))>
  </div>
</Fragment>
`;
